web/src/components/LoginScreen.js

```diff
@@ -0,0 +1,204 @@
+import React from "react";
+import * as Realm from "realm-web";
+import { useRealmApp } from "../RealmApp";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import TextInput from "@leafygreen-ui/text-input";
+import LGCard from "./Card";
+import { uiColors } from "@leafygreen-ui/palette";
+import validator from "validator";
+
+export default function LoginScreen() {
+  const { app, refresh } = useRealmApp();
+  // Toggle between logging users in and registering new users
+  const [mode, setMode] = React.useState("login");
+  const toggleMode = () => {
+    setMode((oldMode) => (oldMode === "login" ? "register" : "login"));
+  };
+  // Keep track of form input state
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  // Keep track of input validation/errors
+  const [error, setError] = React.useState({});
+  // Whenever the mode changes, clear the form inputs
+  React.useEffect(() => {
+    setEmail("nick.larew@mongodb.com");
+    setPassword("password");
+    setError({});
+  }, [mode]);
+
+  const handleLogin = async () => {
+    setError((e) => ({ ...e, password: undefined }));
+    try {
+      await app.logIn(Realm.Credentials.emailPassword(email, password));
+      refresh();
+    } catch (err) {
+      handleAuthenticationError(err, setError);
+    }
+  };
+
+  const handleRegistrationAndLogin = async () => {
+    const isValidEmailAddress = validator.isEmail(email);
+    setError((e) => ({ ...e, password: undefined }));
```

Comment:
```suggestion
    setError((e) => ({ ...e, password: null }));
```

This is a very political opinion, so feel free to ignore this. But I've always been partial to not setting anything as undefined, but rather setting things to null (because it accomplishes the same functionality but it indicates that it was manually set as null, rather than that it was just never defined). Also, it makes sense in context of this code base because we've set things to null in other areas.

Reply:
Yeah I agree here, this was copypasta code from the previous project that uses a different style.